src/main/java/nl/novi/TechItEasy/controllers/TelevisionController.java

```diff
@@ -1,48 +1,53 @@
 package nl.novi.TechItEasy.controllers;
 
-import org.apache.coyote.Response;
+import nl.novi.TechItEasy.dtos.TelevisionOutputDto;
+import nl.novi.TechItEasy.dtos.TelevisionInputDto;
+import nl.novi.TechItEasy.models.Television;
+import nl.novi.TechItEasy.services.TelevisionService;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 import java.net.URI;
-import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 @RestController
 @RequestMapping("/api/televisions")
 public class TelevisionController {
 
-    List<String> televisions = new ArrayList<>();
+    private final TelevisionService televisionService;
+
+    public TelevisionController(TelevisionService televisionService) {
+        this.televisionService = televisionService;
+    }
 
     @GetMapping()
-    public ResponseEntity<List<String>> getAllTelevisions() {
-        return ResponseEntity.ok(televisions);
+    public ResponseEntity<List<TelevisionOutputDto>> getAllTelevisions() {
+        return ResponseEntity.ok(televisionService.getAllTelevisions());
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Object> getTelevision(@PathVariable int id) {
-        return ResponseEntity.ok(televisions.get(id));
+    public ResponseEntity<TelevisionOutputDto> getTelevision(@PathVariable int id) {
+        return ResponseEntity.ok(televisionService.getTelevisionById(id));
     }
 
     @PostMapping()
-    public ResponseEntity<Object> addTelevision(@RequestBody String televisionName) {
-        televisions.add(televisionName);
-        URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}").buildAndExpand(televisions.size()-1).toUri();
-        return ResponseEntity.created(location).build();
+    public ResponseEntity<TelevisionOutputDto> addTelevision(@RequestBody TelevisionInputDto television) {
```

Comment:
Vergeet in je `@`Requestbody niet om ook de `@`Valid annotatie te gebruiken zodat de validatieregels in je input dto ook werkelijk gevalideerd worden